simplefs/simplefs.go

```diff
@@ -219,10 +220,28 @@ func (k *SimpleFS) doCopy(ctx context.Context, srcPath, destPath keybase1.Path)
 	defer dst.Close()
 
 	if src.Type() == keybase1.DirentType_FILE || src.Type() == keybase1.DirentType_EXEC {
-		_, err = io.Copy(dst, src)
+		err = copyWithCancellation(ctx, dst, src)
+		if err != nil {
+			return err
+		}
 	}
 
-	return err
+	return nil
+}
+
+func copyWithCancellation(ctx context.Context, dst io.Writer, src io.Reader) error {
+	for {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+		_, err := io.CopyN(dst, src, 64*1024)
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
```

Comment:
I think the implementation of `Reader` is allowed to choose whether to return `EOF` with the last non-zero-length read, or to return it on the first zero-length read. But either way, I think it's correct for callers to loop until `EOF` like this.